.github/workflows/python-publish.yml

```diff
@@ -0,0 +1,113 @@
+# This workflow will upload a Python Package using Twine when a release is created
+# For more information see: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-python#publishing-to-package-registries
+
+# This workflow uses actions that are not certified by GitHub.
+# They are provided by a third-party and are governed by
+# separate terms of service, privacy policy, and support
+# documentation.
+
+name: Upload Python Package
+
+on:
+  release:
+    types: [published]
+
+permissions:
+  contents: read
+
+jobs:
+  build-ubuntu-2004:
+    runs-on: ubuntu-20.04
+    steps:
+    - uses: actions/checkout@v3
+
+    - name: Install Rust stable
+      uses: actions-rs/toolchain@v1.0.6
+      with:
+          toolchain: stable
+          override: true
+
+    - name: Build and test
+      working-directory: ./lib
+      run: |
+        cargo check
+        cargo build
+        RUST_LOG=hyperon=debug cargo test
+    - name: Install Conan
+      uses: turtlebrowser/get-conan@v1.1
+      with:
+        version: 1.57.0
+
+    - name: Install cbindgen
+      uses: actions-rs/cargo@v1.0.1
+      with:
+        command: install
+        args: cbindgen
+
+    - name: Install Python dependencies
+      run: |
+        python3 -m pip install -e ./python[dev]
+
+    - name: Setup C API build
+      run: |
+        mkdir -p build
+        cd build
+        # specify C compiler as conan could not find it automatically
+        # see https://github.com/conan-io/conan/issues/4322
+        cmake -DCMAKE_C_COMPILER=gcc ..
+
+    - name: Build C API
+      working-directory: ./build
+      run: |
+        set -e
+        make
+        mkdir -p artifacts
+        chrpath -d ./python/*.so  ./c/cargo/*.so
```

Comment:
I would include this step into `CMakeLists.txt` files which build the corresponding libraries.